### PR TITLE
Added ENCODING series argument on TS.CREATE/TS.ADD and global config. Deprecate CHUNK_TYPE global config.

### PR DESCRIPTION
**Pull request by filipecosta90**
As discussed, this is a backwards compatible change. The usage of `CHUNK_TYPE` global configs warns the user that he is using a deprecated setting and should move towards using `ENCODING`.
Before this PR we assumed that not having the uncompressed option on a serie implied a compressed one ( this is true for now with 2 serie encodings but in the future it would make our like harder ). This is now also fixed.

### CODE REVIEW

File: src/redisgears.h [hunk 1]
```diff
@@ -541,13 +541,13 @@ void MODULE_API_FUNC(RedisGears_AddConfigHooks)(BeforeConfigSet before, AfterCon
 #define REDISGEARS_MODULE_INIT_FUNCTION(ctx, name) \
         RedisGears_ ## name = RedisModule_GetSharedAPI(ctx, "RedisGears_" #name);\
         if(!RedisGears_ ## name){\
-            RedisModule_Log(ctx, "warning", "could not initialize RedisGears_" #name "\r\n");\
+            RedisModule_Log(ctx, "warning", "could not initialize RedisGears_" #name);\
```
Comment (danni-m) [inline src/redisgears.h:544]: @filipecosta90 this file comes from geras repo, i don't have an issue with updating it but every time we gonna sync with gears' version we will override this changes.

File: src/tsdb.c [hunk 2]
```diff
@@ -657,12 +658,18 @@ int SeriesCreateRulesFromGlobalConfig(RedisModuleCtx *ctx,
...
+        rules_options &= SERIES_OPT_UNCOMPRESSED;
+
+        TSGlobalConfig.options &= SERIES_OPT_DEFAULT_COMPRESSION;
         CreateCtx cCtx = {
             .retentionTime = rule->retentionSizeMillisec,
             .chunkSizeBytes = TSGlobalConfig.chunkSizeBytes,
             .labelsCount = compactedRuleLabelCount,
             .labels = compactedLabels,
             .options = TSGlobalConfig.options & SERIES_OPT_UNCOMPRESSED,
+            .options = rules_options,
```
Comment (danni-m) [inline src/tsdb.c:672]: @filipecosta90 doesn't this override the previous line?

File: src/tsdb.c [hunk 3]
```diff
@@ -657,12 +658,18 @@ int SeriesCreateRulesFromGlobalConfig(RedisModuleCtx *ctx,
         compactedLabels[labelsCount + 1].value =
             RedisModule_CreateStringPrintf(NULL, "%ld", rule->timeBucket);
 
+        int rules_options = TSGlobalConfig.options;
+        rules_options &= ~SERIES_OPT_DEFAULT_COMPRESSION;
+        rules_options &= SERIES_OPT_UNCOMPRESSED;
+
+        TSGlobalConfig.options &= SERIES_OPT_DEFAULT_COMPRESSION;
```
Comment (danni-m) [inline src/tsdb.c:665]: @filipecosta90 im not sure what's the meaning here.
1. we reset the default compression bit
2. this seems like a no-op to me, if its uncompressed it wouldn't do anything, and if it isnt it wouldn't change it.
3. in line 665 we update the global config as part of a key creation, which is probably not the right place.

File: src/redisgears.h (hunk 1 above)
Comment (filipecosta90) [reply src/redisgears.h:544]: agree. removed this change

File: src/tsdb.c (hunk 3 above)
Comment (filipecosta90) [reply src/tsdb.c:665]: you're right. I missed this after a merge conflict resolution. 
Like we have on master ([link](https://github.com/RedisTimeSeries/RedisTimeSeries/blob/master/src/tsdb.c#L668)) we use uncompressed chunks for rule created series ( keeping the same behavior). 
I've removed the offending lines.

File: src/tsdb.c [hunk 4]
```diff
@@ -660,12 +661,16 @@ int SeriesCreateRulesFromGlobalConfig(RedisModuleCtx *ctx,
         compactedLabels[labelsCount + 1].value =
             RedisModule_CreateStringPrintf(NULL, "%ld", rule->timeBucket);
 
+        int rules_options = TSGlobalConfig.options;
```
Comment (danni-m) [inline src/tsdb.c:664]: @filipecosta90 Can't we  just use `TSGlobalConfig.options`  as is?

**Review by danni-m — DISMISSED** [review]

**Review by danni-m — APPROVED** [review]

**Pull request merged**